pkg/manager/manager.go

```diff
@@ -118,15 +118,27 @@ type Options struct {
 	// for serving prometheus metrics
 	MetricsBindAddress string
 
+	// Functions to all for a user to customize the values that will be injected.
+	NewCache            NewCacheFunc
+	NewClient           NewClientFunc
+	NewDelegatingClient NewDelegatingClientFunc
```

Comment:
hmm... we try to avoid exposing a difference between these two at this level.  What's the usecase for separating them?  It's probably better/clearer to just have `NewClient`, which means `NewDelegatingClient`, and not leak the fact that we have a difference to the user/require the user to have that distinction.

Reply:
The use case is that I want to not use the `DelegatingReader` for my client. I would like my client to use the cache for unstructured types. 

Because there is no other way to override this client this is the only way that I can think of to update it. Thoughts?

Reply:
but there's not really a use for `NewClient` *except* as a step in constructing the delegating client, so we can just elide that and have:

```go
type Options struct {
    ...
    NewClient func(...) client.Client
    ...
}

func defaultNewClient(...) client.Client {
    directClient := newClient(...)
    return &DelegatingClient{
        Writer: directClient,
        Reader: cacheClient,
    }
}
```

Reply:
I was still thinking about them as two things but I think you are completely correct, I will update now!